src/routing.ts

```diff
@@ -230,7 +229,7 @@ export function createLocation(path: () => string): Location {
       return hash();
     },
     get state() {
-      return state();
+      return state() || null;
```

Comment:
Since state could be anything (including things like `0`, `false`, etc.) we should be a little stricter here and only return `null` if `state() == null` (ie. `null` or `undefined`).

Reply:
We can return just `state()`, this returns `undefined` if there is no state.